adding chart for Satisfy - Composer package hosting

anapsix opened the pull request:
Signed-off-by: Anastas Dancha <anapsix@random.io>

<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:
Adding `stable/satisfy` chart.
[Satisfy]( https://github.com/ludofleury/satisfy) - Satis Composer repository manager with a simple web UI.  
It allows to proxy/mirror Composer packages, or hosting one's own, with Git as package source, for example.

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

anapsix committed:
adding chart for Satisfy - Composer package hosting

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
adding self as maintainer

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
adding satisfy.* parameters to README

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
do not wipe out existing instance

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
switching to labels required by REVIEW_GUIDELINES

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
changing resources nil, recommended in comments

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
removing unsupported ingress.path

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
adding icon

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
[incubator/satisfy]: moving to stable

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
[stable/satisfy]: use modern helm labels

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
[stable/satisfy]: switching image pull-by-digest

  - Using image digest in favor of image tag, both are supported.
    When digest is set, it's always used.
  - Updating containerPort from 8080 to 80, inline with latest image update

Signed-off-by: Anastas Dancha <anapsix@random.io>

unguiculus commented on stable/satisfy/values.yaml:
```diff
@@ -0,0 +1,71 @@
+
+## Application settings
+satisfy:
+  repo_name: myrepo
```
Use camel case

anapsix replied:
sure thing!

unguiculus commented on stable/satisfy/templates/deployment.yaml:
```diff
@@ -0,0 +1,108 @@
+                  name: {{ template "satisfy.fullname" . }}
+                  key: SSH_PRIVATE_KEY
+            {{- end }}
+          {{- if .Values.livenessProbe.enabled }}
```
Why do you make probes optional?

anapsix replied:
I like being able to disable probes, as it often help with debugging. I really wish all charts gave that ability out of the box.

unguiculus commented on stable/satisfy/templates/deployment.yaml:
```diff
@@ -0,0 +1,108 @@
+          command:
+            - /bin/sh
+            - -c
+            - >
```
If you do this instead, you don't need the semi-colons:

```suggestion
            - |
```

anapsix replied:
right, thanks for the tip! <3

unguiculus commented on stable/satisfy/values.yaml:
```diff
@@ -0,0 +1,71 @@
+  # if both image.tag and image.digest are present
+  # digest will be used
+  tag: v3.0.4
+  digest: sha256:16a818ab03b6a6bd2625dc9b4760f079fbbddfdd1473994593dfae958c6e7e29
```
Don't specify both, tag and digest. Only one is used anyways. The digest can be used as tag.

anapsix replied:
Since Docker demanded different syntax for digest vs. tag pulls (`imageName@imageDigest` vs. `imageName:imageTag`) I'm using digest explicitly
I've tested with K8s version 1.12, and it expects same syntax as Docker (`imageName@imageDigest`), when using image digest

anapsix committed:
various improvements..

  - using `satisfy.repo_name` in place of `satisfy.repoName`
  - improving init container script
  - updating image digest, commenting out tag, but leaving it there for reference

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
removing trailing space in `values.yaml`

Signed-off-by: Anastas Dancha <anapsix@random.io>

unguiculus commented on stable/satisfy/README.md:
````diff
@@ -0,0 +1,57 @@
+To install the chart with the release name `my-release`:
+
+```console
+$ helm install --name my-release incubator/satisfy
````
incubator -> stable

anapsix replied:
🤦‍♂️

unguiculus commented on stable/satisfy/values.yaml:
```diff
@@ -0,0 +1,71 @@
+satisfy:
+  repoName: myrepo
+  homepage: http://chart-example.local
+  ssh_private_key: ~  # set it via CLI
```
camel case, please!

anapsix committed:
more camelCase, fixing repo link in README

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
better init script & pod checks, note about homepage/fqdn

Signed-off-by: Anastas Dancha <anapsix@random.io>

anapsix committed:
updating image.digest to latest

Signed-off-by: Anastas Dancha <anapsix@random.io>
